src/main/game/gameboard/Tile.java

```
@@ -76,22 +79,29 @@ public void addUnit(Unit unit) {
             return;
         }*/
         units.add(unit);
+        if (containsUnit == false) { containsUnit = true; }
         //unit.setLocation(this.location);
     }
-
     public void addStructure(Structure structure){
-        if(containsStructure()==false){
+        if(this.containsStructure == false){
             this.structure = structure;
+            this.containsStructure = true;
         }
     }
-
     public void addBattleGroup(BattleGroup battleGroup){
         battleGroups.add(battleGroup);
     }
-
     public void addRallyPoint(RallyPoint rallyPoint){
         rallyPoints.add(rallyPoint);
+        if (containsRallyPoint == false) { containsRallyPoint = true; }
     }
+    public void addEffect(AreaEffect effect){
```

should throw an exception if area effect cannot be applied